client/my-sites/themes/helpers.js

```diff
@@ -50,17 +50,16 @@ var ThemesHelpers = {
 	},
 
 	getDetailsUrl: function( theme, site ) {
-		const baseUrl = config.isEnabled( 'manage/themes/details' ) ? '/theme/' : ThemesHelpers.oldShowcaseUrl;
-
-		if ( ! site ) {
-			return baseUrl + theme.id;
+		if ( site && site.jetpack ) {
+			return site.options.admin_url + 'themes.php?theme=' + theme.id;
 		}
 
-		if ( site.jetpack ) {
-			return site.options.admin_url + 'themes.php?theme=' + theme.id;
+		let baseUrl = ThemesHelpers.oldShowcaseUrl + theme.id;
+		if ( config.isEnabled( 'manage/themes/details' ) ) {
+			baseUrl = `/theme/${ theme.id }/details`;
 		}
 
-		return baseUrl + `${ theme.id }/${ site.slug }`;
+		return baseUrl + ( site ? `/${ site.slug }` : '' );
```

Comment:
We have other occurrences dotted around, so let's do them all in one go in a separate PR and track in #4057.